lib/chef/knife/bmcs_server_create.rb

```diff
@@ -170,6 +177,36 @@ def run
         show_value('Node Name', config[:chef_node_name])
       end
 
+      # Return CreateVnicDetails instance
+      def make_vnic_details
+        vnic_details = OracleBMC::Core::Models::CreateVnicDetails.new
+        vnic_details.assign_public_ip = bool_arg(config[:assign_public_ip]) if config[:assign_public_ip]
+        vnic_details.private_ip = config[:private_ip] if config[:private_ip]
+        vnic_details.subnet_id = config[:subnet_id]
+        vnic_details
+      end
+
+      def get_bootstrap_ip(vnic)
+        # TODO: Consider adding a use_private_ip option.
+        bool_arg(config[:assign_public_ip]) == false ? vnic.private_ip : vnic.public_ip
```

Comment:
Note that bootstrapping with the private IP *might* not work until we complete issue https://github.com/oracle/knife-bmcs/issues/4. So, we should make sure to do that as well before we cut a release with this change.